moped-database/migrations/1652877867946_project-list-view-dedup-signal-assets/up.sql

```diff
@@ -0,0 +1,129 @@
+-- public.project_list_view source
+drop view project_list_view;
+
+CREATE OR REPLACE VIEW public.project_list_view
+AS WITH project_person_list_lookup AS (
+         SELECT mpp.project_id,
+            string_agg(DISTINCT concat(mu.first_name, ' ', mu.last_name, ':', mpr.project_role_name), ','::text) AS project_team_members
+           FROM moped_proj_personnel mpp
+             JOIN moped_users mu ON mpp.user_id = mu.user_id
+             JOIN moped_project_roles mpr ON mpp.role_id = mpr.project_role_id
+          WHERE mpp.status_id = 1
+          GROUP BY mpp.project_id
+        ), funding_sources_lookup AS (
+         SELECT mpf_1.project_id,
+            string_agg(mfs.funding_source_name, ', '::text) AS funding_source_name
+           FROM moped_proj_funding mpf_1
+             LEFT JOIN moped_fund_sources mfs ON mpf_1.funding_source_id = mfs.funding_source_id
+          WHERE mpf_1.funding_status_id = 1
+          GROUP BY mpf_1.project_id
+        ), project_type_lookup AS (
+         SELECT mpt.project_id,
+            string_agg(mt.type_name, ', '::text) AS type_name
+           FROM moped_project_types mpt
+             LEFT JOIN moped_types mt ON mpt.project_type_id = mt.type_id AND mpt.status_id = 1
+          GROUP BY mpt.project_id
+        )
+ SELECT mp.project_uuid,
+    mp.project_id,
+    mp.project_name,
+    mp.project_description,
+    mp.project_description_public,
+    mp.ecapris_subproject_id,
+    mp.project_order,
+    mp.current_status,
+    mp.timeline_id,
+    mp.current_phase,
+    mp.end_date,
+    mp.project_length,
+    mp.fiscal_year,
+    mp.capitally_funded,
+    mp.date_added,
+    mp.added_by,
+    mp.is_retired,
+    mp.milestone_id,
+    mp.task_order,
+    COALESCE(mp.status_id, 0) AS status_id,
+    mp.updated_at,
+    ppll.project_team_members,
+    me.entity_name AS project_sponsor,
+    string_agg(DISTINCT me2.entity_name, ', '::text) AS project_partner,
+        CASE
+            WHEN mp.status_id = 0 OR mp.status_id IS NULL THEN NULL::text
+            WHEN mp.status_id = 1 THEN mp.current_phase
+            ELSE mp.current_status
+        END AS status_name,
+    string_agg(task_order_filter.value ->> 'display_name'::text, ','::text) AS task_order_name,
+    mp.contractor,
+    mp.purchase_order_number,
+    COALESCE( -- coalese because this subquery can come back 'null' if there are no component assets
+     (SELECT JSON_AGG(features.feature) -- this query finds any components and those component's features and rolls them up in a JSON blob
+      FROM moped_proj_components components   
+     LEFT JOIN moped_proj_features features 
+       ON (features.project_component_id = components.project_component_id)
+     WHERE TRUE
+       and components.status_id = 1
+       and components.project_id = mp.project_id), '{}'::json) as project_feature, -- close out that coalese; if null, give us a empty json object
```

Comment:
🆕 Here's the bit that collects the component & feature JSON blob based on the project id of the row.